RedHatInsights/compliance-frontend
# feat(AsyncTableTools): RHINENG-11642 - Add useFilterConfig hook

This PR adds the filtering feature the async table implementation and provides serialisation and example configuration for Compliance for it.

It also adds some minor improvements to ease development and debugging of the async table, most notably it allows enabling/disabling the async table via a localStorage entry. This is to make it easier to enable it, and will avoid situations where the `constants.js` file could be accidentally committed.

**How to test:**

1) Open the Reports page
2) Verify that the ReportsTable is still using the current `TableToolsTable` without any change and without a 'insights:compliance:asynctables' localStorage entry.
3) Add a localStorage entry with the key 'insights:compliance:asynctables' and the value "true"
4) Reload the Reports page
5) Verify that now the page uses the async table implementation. (it should show a log in the console, including logging the current table state)
6) Change some filters. Add, change and remove them and verify everything functions correctly
7) Verify that the table stat properly changes
8) Verify that the serialised table state is correct


## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
